### PR TITLE
fix(python): allow empty websocket tool lists

### LINKED ISSUE #1747: Python WebSocketConnector treats an initialized empty tools list as uninitialized
**Describe the bug**

`WebSocketConnector.tools` raises `RuntimeError("MCP client is not initialized")` when a WebSocket MCP server initializes successfully but returns an empty `tools/list` response (`{"tools": []}`). An MCP server can legitimately expose zero tools, so an initialized empty list should be returned as `[]` instead of being treated as the uninitialized state.

The current guard checks `if not self._tools`, which conflates `None` with an empty list.

**To Reproduce**

1. Create a `WebSocketConnector`.
2. Initialize it against a server whose `tools/list` response is `{"tools": []}`.
3. Read `connector.tools`.
4. The property raises `RuntimeError` even though initialization completed.

Minimal local repro used for verification:

```python
from typing import Any

import pytest

from mcp_use.client.connectors.websocket import WebSocketConnector

@pytest.mark.asyncio
async def test_tools_property_allows_initialized_empty_tool_list():
    connector = WebSocketConnector("ws://localhost:8080")

    async def fake_send_request(method: str, params: dict[str, Any] | None = None) -> dict[str, Any]:
        if method == "initialize":
            return {}
        if method == "tools/list":
            return {"tools": []}
        raise AssertionError(f"unexpected method: {method}")

    connector._send_request = fake_send_request

    await connector.initialize()

    assert connector.tools == []
```

**Expected behavior**

After successful initialization, `connector.tools` should return the empty list from the server. Only the `None` uninitialized sentinel should raise.

**Additional context / verification**

I verified this locally in `libraries/python` with a focused unit test. With the guard changed from `if not self._tools` to `if self._tools is None`, the verification command passes:

```bash
MCP_USE_ANONYMIZED_TELEMETRY=false .venv/bin/python -m pytest tests/unit/test_websocket_connector.py -q
```

Result: `1 passed`.

### PR DESCRIPTION
**Pull request by luochen211**
## Summary
- treat an empty initialized WebSocket tool list as valid
- keep the RuntimeError only for the uninitialized `_tools is None` state
- add a regression test for WebSocket servers that expose zero tools

## Tests
- MCP_USE_ANONYMIZED_TELEMETRY=false .venv/bin/python -m pytest tests/unit/test_websocket_connector.py -q
- uv run --extra dev pytest tests/unit/test_websocket_connector.py tests/unit/test_websocket_connection_manager.py -q
- uv run --extra dev ruff check mcp_use/client/connectors/websocket.py tests/unit/test_websocket_connector.py
- uv run --extra dev ruff format --check mcp_use/client/connectors/websocket.py tests/unit/test_websocket_connector.py

Closes #1747

### CODE REVIEW

File: libraries/python/pyproject.toml [hunk 1]
```diff
@@ -20,7 +20,7 @@ classifiers = [
     "Topic :: Software Development :: Libraries :: Python Modules",
 ]
 dependencies = [
-    "mcp>=1.24.0",
+    "mcp>=1.24.0,<2.0.0",
```
[the remark this answers is not in the document]
Comment (luochen211) [reply libraries/python/pyproject.toml:23]: Fixed in af2150626. DEPENDENCY_POLICY.md now records the exact supported range mcp>=1.24.0,<2.0.0, explains that MCP SDK 2.x has breaking APIs not yet migrated, and states that the cap should be removed as part of a validated 2.x migration.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -269,6 +269,7 @@ jobs:
           path: test-results
           merge-multiple: false
       - name: Generate PR comment
+        continue-on-error: ${{ github.event.pull_request.head.repo.fork }}
```
[the remark this answers is not in the document]
Comment (luochen211) [reply .github/workflows/ci.yml]: Fixed in ad32b44d1. Fork PRs now skip both CI report-comment steps entirely instead of running guaranteed-failing writes. The same skip condition is combined with the artifact-success checks for both conformance comment steps; internal PR comment failures remain visible.

**Comment by khandrew1** [comment]
Thanks for taking the time to contribute. Our focus right now is on the TypeScript SDK, so we're limiting Python PRs to fixes for concrete problems that contributors have run into firsthand. This PR doesn't appear to address that kind of issue, so we're closing it for now, along with the linked issue. If this was prompted by something you hit in practice, please reply on the issue with what you were doing when it came up and how to reproduce it, and we'll reopen the issue and take another look.
